Dev2.0 angel bob compatibility

Sectoid opened the pull request:
Hello,

Here are some adaptions of Angel's mods for the recent Bob's mods changes (mostly for 'bob-' prefix). The changes are not complete and are not (yet?) tested with a playthrough. 

This is my first changes for Factorio mods, so looking forward for any comments/suggestions/etc.

Sectoid committed:
angelsmelting: initial mod update

it loads!

Sectoid committed:
angeladdons-cab: loads!

Sectoid commented:
Angel's Bioprocessing and Angel's Industry is not covered yet, to be updated soon.

KiwiHawk commented:
Excellent work! Thank you for this. I'm very nearly done with Bob's. Once I've got them released I'll switch focus to Angel's and get this reviewed.

Sectoid committed:
angelsbioprocessing: initial mod update

Updated to support 'bob-' prefixed changes. Successfully loads,
playthrough untested.

modules tiering assumption:
  modules 1-2 -> tier1
  modules 3-4 -> tier2
  modules 5   -> tier3

TODO:
  * green modules overhaul (disabled for now)
  * quality support

Sectoid commented:
Initial Angel's Bioprocessing porting done. Still TODO how to properly overhaul Green modules. They seem to be newly added, so looking for help on these ones.

KiwiHawk commented:
No, Green modules aren't new. A couple of module types have been removed even: Pure productivity, Pure speed.

Sectoid committed:
angelsindustries: initial mod update

components & science overhauls completely untested

Sectoid commented:
I've added green modules, and also (partially?) fixed alien artifacts processing. However I haven't managed to properly change angelsbioprocessing/prototypes/technology/bio-processing-artifacts.lua, bcs it uses `alien-artifact-*-tool` things as science pack, and Factorio complains there are no lab that can accept those. And I agree with it -- I haven't managed to find one in the code either :) Maybe I'm missing something or lack knowledge to understand how this should work. Looking to the 1.1 mod didn't help either.

Also I made initial update of Angel's Industry. It even loads a new game... :)

Sectoid committed:
angelsbioprocessing: small fixes

* adjust alien artifacts names with 'bob-' prefix in loot events
* disable bob's wood pellets
* fix renamed bob-coal-from-wood recipe. maybe this one should be
  disabled completely? it gives access to carbon too early

Sectoid commented:
I've tried to make a test playthrough, fixed some minor bugs. Quite soon I've found out that gardens/puffers/other stuff doesn't get spawned on the map. Looks like something is broken in autoplace settings, but I'm too dumb to understand all that NoiseExpressions documentation. Probably I will need some help from comunity.

Btw, do you have a Discord server or some other way to communicate to modpack devs community?

KiwiHawk commented:
Angel's discord: https://discord.gg/ff5p6KE

Sectoid committed:
angelsbioprocessing: fixed autoplacement of fish, gardens, trees and puffer nests

Sectoid committed:
angelsindustries: adjust bob's equipment and vehicle equiment techs

Sectoid committed:
angelsindustries: fix angels-components category icon

Sectoid committed:
angelssmelting: gilded-copper-cable and intergrated-electronics fix

Sectoid committed:
angelssmelting: fix fluid boxes for casting-machine and induction-furnace

+ forced symmetry (graphics still to be done)

Sectoid committed:
angelssmelting: fix fluid_boxes of cooling-tower

Sectoid committed:
angelssmelting: fix fluid boxes for blast-furnace

Sectoid committed:
angelssmelting: fix molten smelting recipes icons sizes

Sectoid committed:
add forced flipping to entities (#972)

looks ugly but playable

Sectoid committed:
angelsindustries: remove green modules

KiwiHawk committed:
Merge branch 'dev2.0' into pr/1027

KiwiHawk committed:
Merge branch 'dev2.0' into pr/1027

KiwiHawk committed:
Merge branch 'dev2.0' into pr/1027

KiwiHawk committed:
Fix merge error

KiwiHawk committed:
Merge branch 'dev2.0' into pr/1027

KiwiHawk committed:
Revert unnecessary changes

KiwiHawk committed:
Merge branch 'dev2.0' into pr/1027

KiwiHawk committed:
Reverting changes

- Industries mod will be fixed separately
- Nuclear fuel cells have been fixed a different way
- Equipment categories seem fine without this?
